Alluvial.Tests/StreamCatchupTests.cs

```diff
@@ -278,82 +212,46 @@ await StreamCatchup.Create(streamSource.UpdatedStreams())
             var projector = new BalanceProjector()
                 .Pipeline(async (projection, batch, next) =>
                 {
-                    if (projectionStore.Count() == 30)
-                    {
-                        throw new Exception("oops");
-                    }
+                    Throw();
                     await next(projection, batch);
                 })
                 .Catch(continueIf: (projection, batch, next) => true);
 
-            var catchup = StreamCatchup.Create(streamSource.UpdatedStreams(), batchCount: 50)
+            var catchup = StreamCatchup.Create(stream, batchCount: 50)
                                        .Subscribe(projector, projectionStore);
 
             await catchup.RunSingleBatch();
 
-            projectionStore.Count().Should().Be(50);
+            projectionStore.Count().Should().Be(1);
         }
 
         [Test]
-        public async Task Catchup_cursor_storage_can_be_specified_using_catchup_configuration()
+        public async Task Catchup_can_query_streams_such_that_repeated_data_is_not_queried()
         {
-            ICursor storedCursor = null;
-
-            var catchup = StreamCatchup.Create(streamSource.UpdatedStreams(),
-                                               batchCount: 1,
-                                               configure: c => c.StoreCursor(async (id, cursor) => storedCursor = cursor))
-                                       .Subscribe(new BalanceProjector());
-
-            var query = await catchup.RunSingleBatch();
-
-            storedCursor.Should().BeSameAs(query.Cursor);
-        }
-
-        [Test]
-        public async Task Catchup_cursor_retrieval_can_be_specified_using_catchup_configuration()

```

Comment:
In the case of a single-stream catchup, the projection is the cursor, so it's not really necessary to pull this out as a separate concept. I'm going to leave this out of the interface for now and bring it back if we find it's needed.